actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/LearnerStateActor.java

```diff
@@ -440,4 +445,61 @@ private String generatePrimaryKeyForContent(
             + batchId;
     return OneWayHashing.encryptVal(key);
   }
+
+  private List<Map<String,Object>> calculateProgressForUserCourses(Request request, Map<String, Object> result)
+  {
+    List<Map<String,Object>> activeCourses = (List<Map<String,Object>>)(result.get(JsonKey.CONTENT));
+    List<Map<String, Object>> contentsForCourses = getcontentsForCourses(request,activeCourses);
+    List<Map<String,Object>> updatedCourses = new ArrayList<>();
+    int progress=0;
+    for(Map<String,Object> course : activeCourses) {
+      UserCourses userCourseResult = userCourseDao.read((String) course.get(JsonKey.BATCH_ID), (String) course.get(JsonKey.USER_ID));
+        if(userCourseResult !=null && userCourseResult.getProgress()==100)
```

Comment:
use status column to check wherever necessary, content_consumption should be used to validate the progress